### ISSUE TITLE
Can one use the Newton solver with compositing material model?

### ISSUE THREAD

**Issue by elodie-kendall** [geodynamics/aspect#4388]
Hi all,

I was tackling some convergence errors when increasing the maximum viscosity in my global spherical shell models. I managed to get the residual down massively by increasing the resolution and lowering the cfl. I attach the residual with time, suggesting that the defect correction or newton solver would be useful. I'm confused why the defect correction stokes (I also reduced the linear tolerance to 1e-2) would perform worse and even fail with the attached solver fail. Secondly, the newton solver is also failing (attachted) on timestep 0! on the switch from the Picard to Newton scheme. Is this because MaterialModelDerivatives is not computed for the newton solver with the compositing material model?

I also attach my prm. I would really appreciate some help as soon as you can.

Thanks,
Elodie

<img width="1215" alt="residual" src="https://user-images.githubusercontent.com/75854106/139449310-e23b455d-3e8f-4412-a285-85cf54d446ac.png">

<img width="1586" alt="defect_corr_stokes" src="https://user-images.githubusercontent.com/75854106/139449290-b058fde9-11d2-476e-90a6-d7347ed3e497.png">

<img width="1613" alt="newton" src="https://user-images.githubusercontent.com/75854106/139449782-38f98653-53ce-4f8d-beff-d6f1e22330af.png">

**Comment by elodie-kendall** [comment]
Thanks for your reply Meno! I tried with Nonlinear solver scheme = iterated Advection and defect correction Stokes and the advection solver is still failing to converge- see screenshot
<img width="1603" alt="Screen Shot 2021-11-01 at 10 12 52" src="https://user-images.githubusercontent.com/75854106/139649625-2bec5906-9752-47a5-97a9-957fdd2d001d.png">

**Comment by anne-glerum** [comment]
Hi @MFraters, to clarify: with the `single Advection, iterated Stokes` solver, the model runs to completion. The same setup with `single Advection, iterated defect correction Stokes` fails in a temperature solve, and with `single Advection, iterated Newton Stokes` fails when switching from Picard to Newton for the first time.

**Comment by elodie-kendall** [comment]
Yes thanks, Anne. And for the case where "single Advection, iterated defect correction Stokes" fails velocities are only mm/yr and I don't see some issues in T either.

<img width="977" alt="Screen Shot 2021-11-03 at 11 18 20" src="https://user-images.githubusercontent.com/75854106/140043403-227dd5bb-3cc9-45b8-886d-cea24ed8080e.png">

<img width="973" alt="Screen Shot 2021-11-03 at 11 20 35" src="https://user-images.githubusercontent.com/75854106/140043526-bf5c9fc2-96de-4eb2-bbd2-8cca13eb8a82.png">

**Comment by MFraters** [comment]
Ah, thanks for the clarification. This is really strange. Is there any difference in the temperature and/or velocity field when using the `single Advection, iterated Stokes` versus the `single Advection, iterated defect correction Stokes`? Also, can the issue be reproduced on a lower resolution grid?

@tjhei or @bangerth any idea's how the choice of stokes solver can cause the advection solver to crash (they should both yield the same velocity field)?

**Comment by elodie-kendall** [comment]
Hi Menno. Thanks for your reply:) Actually, I see the differences in convergence straight away- on the last iteration of Timestep 0, "iterated stokes" can converge but "iterated defect correction Stokes" can't. The T profiles look the same but the velocities are different- see attached. 

<img width="1148" alt="Screen Shot 2021-11-08 at 14 12 44" src="https://user-images.githubusercontent.com/75854106/140748202-d3927b1b-eb49-445d-9821-5635681d64fe.png">

<img width="1145" alt="Screen Shot 2021-11-08 at 14 12 57" src="https://user-images.githubusercontent.com/75854106/140748190-5f31da5f-0fe4-45cc-87ac-79ad611cd34e.png">

**Comment by elodie-kendall** [comment]
Hi Menno. For the same tolerances (lin and non lin=1e-5) iterated and defect CS give different velocities. I removed complexity from this prm one at a time and only found convergence with the defect CS when I changed from compositing (viscosity and reaction term = visco plastic but all other fields = ascii ref profile) to fully viscoplastic. So there is some problem with the compressibility and the defect CS/Newton solver
